Tasks/JavaToolInstallerV0/javatoolinstaller.ts

```diff
@@ -82,22 +73,142 @@ async function getJava(versionSpec: string) {
             console.log(taskLib.loc('RetrievingJdkFromLocalPath'));
             jdkFileName = taskLib.getInput('jdkFile', true);
         }
-        // unpack the archive, set `JAVA_HOME` and save it for further processing
-        compressedFileExtension = JavaFilesExtractor.getFileEnding(jdkFileName);
-        extractDirectoryName = `${extendedJavaHome}_${JavaFilesExtractor.getStrippedName(jdkFileName)}_${compressedFileExtension.substr(1)}`;
-        extractionDirectory = path.join(extractLocation, extractDirectoryName);
-        await unpackArchive(extractionDirectory, jdkFileName, compressedFileExtension);
-        jdkDirectory = JavaFilesExtractor.setJavaHome(extractionDirectory);
+        compressedFileExtension = JavaFilesExtractor.getSupportedFileEnding(jdkFileName);
+        jdkDirectory = await installJDK(jdkFileName, compressedFileExtension, extractLocation, extendedJavaHome, versionSpec, cleanDestinationDirectory);
     }
     console.log(taskLib.loc('SetExtendedJavaHome', extendedJavaHome, jdkDirectory));
     taskLib.setVariable(extendedJavaHome, jdkDirectory);
     toolLib.prependPath(path.join(jdkDirectory, BIN_FOLDER));
 }
 
-function sleepFor(sleepDurationInMillisecondsSeconds): Promise<any> {
-    return new Promise((resolve, reeject) => {
-        setTimeout(resolve, sleepDurationInMillisecondsSeconds);
-    });
+/**
+ * Install JDK.
+ * @param sourceFile Path to JDK file.
+ * @param fileExtension JDK file extension.
+ * @param archiveExtractLocation Path to folder to extract a JDK.
+ * @returns string
+ */
+async function installJDK(sourceFile: string, fileExtension: string, archiveExtractLocation: string, extendedJavaHome: string, versionSpec: string, cleanDestinationDirectory: boolean): Promise<string> {
+    let jdkDirectory;
+    if (fileExtension === '.dmg' && os.platform() === 'darwin') {
+        // Using set because 'includes' array method requires tsconfig option "lib": ["ES2017"]
+        const volumes: Set<string> = new Set(fs.readdirSync(VOLUMES_FOLDER));
+
+        await attach(sourceFile);
+    
+        const volumePath: string = getVolumePath(volumes);
+
+        let pkgPath: string = getPackagePath(volumePath);
+        try {
+            jdkDirectory = await installPkg(pkgPath, extendedJavaHome, versionSpec);
+        } catch (error) {
+            throw error;
```

Comment:
Is catch required here? Could we just omit it?

Reply:
Fair note, thanks! Removed it